### ISSUE TITLE
JOSS paper review - Installation docs 

### ISSUE THREAD

**Issue by kjappelbaum** [sparks-baird/xtal2png#184]
- [x] Installation instructions 

> There are two conflicting sets of install instructions on the README: conda create -n xtal2png -c conda-forge xtal2png m3gnet at the top and conda env create -n xtal2png -c conda-forge xtal2png in the Installation section. The inclusion of m3gnet results in a very long (possibly infinite at the time of writing 😉) environment solving time. I understand m3gnet is optional, which is explained, but it might be worth not including this in the first installation instruction to save the user some frustration.

- [x] Example 
> The Example in the README doesn't work straight away for me. I think because XtalConverter is in xtal2png.core now, not in xtal2png. Similarly example_structures is now in xtal2png.utils.data.

from https://github.com/openjournals/joss-reviews/issues/4528#issuecomment-1180113793

**Comment by kjappelbaum** [comment]
Running the example actually produces the following 

```python
In [9]: for i, s in enumerate(example_structures):
   ...:     print(s.get_space_group_info())
   ...: 
('Pccn', 56)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [9], in <cell line: 1>()
      1 for i, s in enumerate(example_structures):
----> 2     print(s.get_space_group_info())

File ~/miniconda3/envs/xtal2png/lib/python3.8/site-packages/pymatgen/core/structure.py:1013, in IStructure.get_space_group_info(self, symprec, angle_tolerance)
   1010 from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
   1012 a = SpacegroupAnalyzer(self, symprec=symprec, angle_tolerance=angle_tolerance)
-> 1013 return a.get_space_group_symbol(), a.get_space_group_number()

File ~/miniconda3/envs/xtal2png/lib/python3.8/site-packages/pymatgen/symmetry/analyzer.py:97, in SpacegroupAnalyzer.get_space_group_symbol(self)
     90 def get_space_group_symbol(self):
     91     """
     92     Get the spacegroup symbol (e.g., Pnma) for structure.
     93 
     94     Returns:
     95         (str): Spacegroup symbol for structure.
     96     """
---> 97     return self._space_group_data["international"]

TypeError: 'NoneType' object is not subscriptable
```

**Comment by kjappelbaum** [comment]
apparently, because `spglib.get_symmetry_dataset` returns `None` for this structure.

**Comment by dandavies99** [comment]
Hi @kjappelbaum, @sgbaird - I'm still getting import errors trying the example as in the README. I think it should be:

```diff
- from xtal2png import XtalConverter, example_structures
+ from xtal2png.core import XtalConverter
+ from xtal2png.utils.data import example_structures
```

**Comment by kjappelbaum** [comment]
thanks!

did you install from the `HEAD` or did you test some specific release?  In principle, those things should be exported here ... https://github.com/sparks-baird/xtal2png/blob/464e767207ee281fc19cfaa12d87f17ff6b988c4/src/xtal2png/__init__.py#L20-L21 

Anyhow, let's add a test for this example.

**Comment by sgbaird** [comment]
Strange, I would have thought it would work based on https://github.com/sparks-baird/xtal2png/blob/main/src/xtal2png/__init__.py#L20-L21. Which version are you using and how are you doing the installation?

**Comment by dandavies99** [comment]
Ah yes you're right it should import fine. I installed using conda (without m3gnet), and the `__init__.py` doesn't have the public API section:

```python
import sys

if sys.version_info[:2] >= (3, 8):
    # TODO: Import directly (no need for conditional) when `python_requires = >= 3.8`
    from importlib.metadata import PackageNotFoundError, version  # pragma: no cover
else:
    from importlib_metadata import PackageNotFoundError, version  # pragma: no cover

try:
    # Change here if project is renamed and does not equal the package name
    dist_name = __name__
    __version__ = version(dist_name)
except PackageNotFoundError:  # pragma: no cover
    __version__ = "unknown"
finally:
    del version, PackageNotFoundError
```

**Comment by sgbaird** [comment]
@dandavies99 ah, right. This should be corrected in `v0.8.0`, but conda-forge is still at `v0.7.0` while @kjappelbaum is working on getting `element-coder` up on conda-forge. I'll change the directions back to what you suggested in the meantime, and then change it back once `conda-forge` is up to `v0.8.0`.

**Comment by dandavies99** [comment]
Great that makes sense! It's a v minor point, but just thinking about a novice user playing with the code for the first time; they'll probably install using conda then try and run the first example, so good to have those things consistent 😄
